### PR TITLE
Uniform circleCI and build_docs workflows across repositories

### LINKED ISSUE #261: Uniform circleci workflow with main repository
## 🧰 Task

In the context of the fail spotted in #259 I found that circleci workflow between the main repository and this repository differs. 

I think that we should find a way to uniform it for simplifaction of maintenance.

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues
together with https://github.com/napari/napari/pull/6417 closes #261

# Description

Diff:
```diff
diff .circleci/config.yml ../napari/.circleci/config.yml                                                     1 ↵
10c10
<           path: docs
---
>           path: napari
12,13c12,13
<           name: Clone main repo into a subdirectory
<           command: git clone git@github.com:napari/napari.git napari
---
>           name: Clone docs repo into a subdirectory
>           command: git clone git@github.com:napari/docs.git docs
```

```diff
diff .github/workflows/build_docs.yml ../napari/.github/workflows/build_docs.yml                             1 ↵
2c2
< # https://github.com/napari/napari/blob/main/.github/workflows/build_docs.yml
---
> # https://github.com/napari/docs/blob/main/.github/workflows/build_docs.yml
6c6
<   pull_request:
---
>   push:
8c8,10
<       - main
---
>       - docs
>     tags:
>       - 'v*'
9a12
>   workflow_call:
19a23
>           repository: napari/docs
25d28
<           repository: napari/napari
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,74 +1,48 @@
...
+            python -m pip install -e "napari/[pyside,dev]"
+          environment:
+            PIP_CONSTRAINT: napari/resources/constraints/constraints_py3.10_docs.txt
       - run:
           name: Build docs
           command: |
             . venv/bin/activate
-            xvfb-run --auto-servernum make docs GALLERY_PATH=../napari/examples/
+            cd docs
+            xvfb-run --auto-servernum make docs
```
Comment (melissawm) [inline .circleci/config.yml:43]: Same comment as napari/napari#6417, check if we need to explicitly set GALLERY_PATH

**Review by melissawm — APPROVED** [review]
LGTM!

File: .circleci/config.yml (hunk 1 above)
Comment (Czaki) [reply .circleci/config.yml:43]: The `GALLERY_PATH` should be `GALLERY_PATH=../../napari/examples/` but this is default value

**Comment by psobolewskiPhD** [comment]
Same as the other PR, I'd like to keep (some) of the CircleCI comments, just to make life easier from a maintenance PoV.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,74 +1,53 @@
 # Use the latest 2.1 version of CircleCI pipeline process engine.
```
Comment (psobolewskiPhD) [inline .circleci/config.yml:3]:
```suggestion
# As much as possible, this file should be kept in sync with:
# https://github.com/napari/napari/blob/main/.circleci/config.yaml
# Use the latest 2.1 version of CircleCI pipeline process engine.
```

**Review by psobolewskiPhD — APPROVED** [review]
Small comment made for both PR to add reminder about syncing the circleCI too.

**Comment by Czaki** [comment]
merge as https://github.com/napari/napari/pull/6417 is merged

**Pull request merged**